### ISSUE TITLE
compliance: 45 remaining false negatives across §6.1/6.2/6.3/6.4/6.6/6.7 clauses

### ISSUE THREAD

**Issue by jasperdew** [jasperdew/xfa-native-rust#456]
## Summary
After implementing targeted fixes, the compliance checker still misses 45 violations across many ISO 19005-2 clauses. These are categorized below for parallel implementation.

## Impact
45 false negatives in 4K corpus (run-20260315-065448.905). Combined with #453 and #454 this totals 85 compliance false negatives.

### §6.2.x — Font checks (20 PDFs)
```
§6.2.2 (Undefined content stream operators):
  stressful/batch1/PDFBOX/PDFBOX-1760-11.pdf
  stressful/batch3/MOZILLA/MOZILLA-869065-0.pdf
  stressful/batch2/GHOSTSCRIPT/GHOSTSCRIPT-698804-0.pdf → ["6.2.3.2"]

§6.2.3.2 (Font /Widths):
  tagged/verapdf/veraPDF test suite 6-2-3-2-t01-undefined-a.pdf
  stressful/batch1/PDFBOX/PDFBOX-1760-11.pdf

§6.2.4.2 (Font encoding):
  tagged/verapdf/veraPDF test suite 6-2-4-2-t03-fail-c.pdf
  tagged/verapdf/veraPDF test suite 6-2-4-2-t03-fail-f.pdf

§6.2.4.3 (Font encoding differences):
  stressful/batch2/GHOSTSCRIPT/GHOSTSCRIPT-696802-1.pdf

§6.2.4.4 (Font encoding):
  tagged/verapdf/veraPDF test suite 6-2-4-4-t03-fail-d.pdf

§6.2.7.1 (Font program):
  tagged/verapdf/veraPDF test suite 6-2-7-1-t02-fail-a.pdf

§6.2.8.3 (Type3 font):
  tagged/verapdf/veraPDF test suite 6-2-8-3-t03-fail-a.pdf
  tagged/verapdf/veraPDF test suite 6-2-8-3-t04-fail-a.pdf

§6.2.9 (Font /Encoding):
  tagged/verapdf/veraPDF test suite 6-2-9-3-t01-fail-a.pdf

§6.2.10.x (CIDFont metrics):
  tagged/verapdf/veraPDF test suite 6-2-10-3-1-t01-fail-a.pdf → ["6.2.10.3.1"]
  tagged/verapdf/veraPDF test suite 6-2-10-3-2-t01-fail-c.pdf → ["6.2.10.3.2"]
  tagged/verapdf/veraPDF test suite 6-2-10-4-1-t02-fail-c.pdf → ["6.2.10.9","6.2.10.7","6.2.10.5"]
  tagged/verapdf/veraPDF test suite 6-2-10-5-t01-fail-g.pdf   → ["6.2.10.5"]
  tagged/verapdf/veraPDF test suite 6-2-10-8-t01-fail-a.pdf   → ["6.2.10.8"]

§6.2.11.5/6/7.2/7.3 (CIDFont ToUnicode):
  stressful/batch2/GHOSTSCRIPT/GHOSTSCRIPT-694769-1.pdf → ["6.2.11.5","6.2.10"]
  stressful/batch2/GHOSTSCRIPT/GHOSTSCRIPT-696802-1.pdf → ["6.2.4.3","6.2.10"]
  tagged/verapdf/veraPDF test suite 6-2-11-6-t02-fail-a.pdf   → ["6.2.11.6"]
  tagged/verapdf/veraPDF test suite 6-2-11-7-2-t01-fail-b.pdf → ["6.2.11.7.2"]
```

### §6.1.x — Low-level structure (5 PDFs)
```
§6.1.7 (Stream Length mismatch):
  general/govdocs/276_276437.pdf → ["6.1.7"]
  stressful/batch2/GHOSTSCRIPT/GHOSTSCRIPT-688790-4.pdf → ["6.1.7"]

§6.1.8 (Character encoding):
  stressful/batch2/GHOSTSCRIPT/GHOSTSCRIPT-688790-4.pdf → ["6.1.8"]

§6.1.12 (Document ID):
  tagged/verapdf/isartor-6-1-12-t01-fail-b.pdf → ["6.1.12"]

§6.1.13 (Trailer /Encrypt not allowed):
  tagged/verapdf/TWG test suite A018-pdfa2-fail-a.pdf → ["6.1.13"]
  tagged/verapdf/TWG test suite A018-pdfa2-fail-b.pdf → ["6.1.13"]

§6.1.3 (Xref offset padding):
  tagged/verapdf/veraPDF test suite 6-1-3-t01-fail-b.pdf → ["6.1.3"]
  tagged/verapdf/veraPDF test suite 6-1-3-t05-fail-a.pdf → ["6.1.3"]
§6.1.7.1 (Stream /Length):
  tagged/verapdf/veraPDF test suite 6-1-7-1-t01-fail-a.pdf → ["6.1.7.1"]
§6.1.8 (Char encoding):
  tagged/verapdf/veraPDF test suite 6-1-8-t01-fail-d.pdf → ["6.1.8"]
```

### §6.2.2 — Undefined operators (3 PDFs)
```
tagged/verapdf/veraPDF test suite 6-2-2-t02-fail-c.pdf → ["6.2.2"]
tagged/verapdf/veraPDF test suite 6-2-2-t03-fail-b.pdf → ["6.2.2"]
stressful/batch1/PDFBOX/PDFBOX-3105-1.pdf → ["6.2.2"]
```

### §6.3.x — Colorspace (5 PDFs)
```
tagged/verapdf/isartor-6-3-2-t01-fail-c.pdf → ["6.3.5","6.3.4"]
tagged/verapdf/isartor-6-3-6-t01-fail-a.pdf → ["6.3.6"]
tagged/verapdf/isartor-6-3-7-t02-fail-a.pdf → ["6.3.7"]
tagged/verapdf/veraPDF test suite 6-3-5-t03-fail-c.pdf → ["6.3.5"]
tagged/verapdf/veraPDF test suite 6-3-2-t01-fail-y.pdf → ["6.3.2"]
stressful/batch2/GHOSTSCRIPT/GHOSTSCRIPT-688790-4.pdf → ["6.3.5","6.3.4"]
stressful/batch5/poppler/poppler-106863-0.pdf → ["6.3.6"]
```

### §6.4.3 / §6.6.1 / §6.6.2 (4 PDFs)
```
stressful/batch2/GHOSTSCRIPT/GHOSTSCRIPT-699313-2.7z-1.pdf → ["6.4.3"]
tagged/verapdf/veraPDF test suite 6-4-1-t02-fail-a.pdf → ["6.6.1"]
tagged/verapdf/veraPDF test suite 6-4-2-t01-fail-a.pdf → ["6.6.2"]
invoices/zugferd/ZTESTZUGFERD_1_INVDSS_012015738820PDF-1.pdf → ["6.6.4","6.2.11.4.2"]
```

### §6.7.9 (additional) / §6.2.3.2 (1 PDF)
```
stressful/batch1/PDFBOX/PDFBOX-2525-0.pdf → ["6.7.9"]
tagged/verapdf/veraPDF test suite 6-5-1-t01-fail-c.pdf → ["6.3.1"]
```

## Fix location
`crates/pdf-compliance/src/check.rs` — implement missing clause checks per group above.
